__tests__/app.test.js

```diff
@@ -155,5 +156,91 @@ describe("/api/articles/:article_id/comments", () => {
 			expect(body.msg).toBe('Article does not exist');
 		  });
 	  });
+	test('POST: 201 - adds a new comment for the specific article and sends the new comment back as a response', () => {
+		const newComment = {
+		  username: 'rogersop',
+		  body: "It's amazing"
+		};
+		return request(app)
+		  .post("/api/articles/3/comments")
+		  .send(newComment)
+		  .expect(201)
+		  .then(({body}) => {
+			expect(typeof body.comment.comment_id).toBe("number");
+			expect(body.comment.article_id).toBe(3);
+			expect(body.comment.author).toBe('rogersop');
+			expect(body.comment.body).toBe("It's amazing");
+		  });
+	  });
+	  test('POST: 201 - ignores additional keys in the given object and creates the comment as long as the necessary keys are included', () => {
+		const newComment = {
+		  username: 'rogersop',
+		  body: "It's amazing",
+		  author: "Pikachu"
+		};
+		return request(app)
+		  .post("/api/articles/3/comments")
+		  .send(newComment)
+		  .expect(201)
+		  .then(({body}) => {
+			expect(typeof body.comment.comment_id).toBe("number");
+			expect(body.comment.article_id).toBe(3);
+			expect(body.comment.author).toBe('rogersop');
+			expect(body.comment.body).toBe("It's amazing");
+		  });
+	  });
+	  test("POST: 404 - responds with an appropriate status and error message when provided with the article id that doesn't exist", () => {
+		const newComment = {
+			username: 'rogersop',
+			body: "It's amazing"
+		  };
+		return request(app)
+		  .post('/api/articles/99/comments')
+		  .send(newComment)
+		  .expect(404)
+		  .then(({body}) => {
+			expect(body.msg).toBe('Article does not exist');
+		  });
+	  });
+	  test("POST: 406 - responds with an appropriate status and error message when provided with the username that doesn't exist", () => {
```

Comment:
I tend to think of this as a 404, but can see the argument for viewing it as a 406 also